[ADD] crm_claim_type: added module to clasify the crm_claims

nhomar opened the pull request:
This module was extracted from vauxoo/rma for vauxoo/crm and called crm_claim_type following suggestions listed https://github.com/OCA/rma/pull/35#issuecomment-128056338

Commit:
[ADD] New module crm_claim_type that allows to register claim types

Commit:
[REF] crm_claim_type: modify help text for claim type field

Commit:
[FIX] crm_claim_type: move security to data, is not valid for dictionary

Commit:
[ADD] crm_claim_type: i18n for es, es_VE, es_MX and es_PA

Commit:
[REF] crm_claim_type: removed strings from model's fields

Commit:
[REF] crm_claim_type: added license header

Commit:
[REF] crm_claim_type: replace project_repo, version and module_name

Commit:
[REF] crm_claim_type: add license, remove description and add OCA as author

Commit:
[REF] crm_claim_type: remove string from model's field

Commit:
[REF] crm_claim_type: add module version

nhomar committed:
Merge pull request #1 from vauxoo-dev/8.0-crm_claim_type-dev-osval

8.0 crm claim type dev osval

pedrobaeza commented on crm_claim_type/README.md:
```diff
@@ -0,0 +1,61 @@
+CRM Claim Types
+==============
```
Put here the same length as the title

pedrobaeza commented on crm_claim_type/README.md:
```diff
@@ -0,0 +1,61 @@
+CRM Claim Types
+==============
+
+Includes a way to clasify claims adding the concept of type.
```
s/clasify/classify

pedrobaeza commented on crm_claim_type/README.md:
```diff
@@ -0,0 +1,61 @@
+
+This modules do not require special configurations.
+It contains a predefined set of claim types.
+But if you want add your own types go to Sales > Configurations > Claims > Claim Types
```
s/want add/want to add

pedrobaeza commented on crm_claim_type/README.md:
```diff
@@ -0,0 +1,61 @@
+
+To use this module, you need to:
+
+* Go to Sales > Configurations > Claims > Claim Types
```
Here you have to say that you need to go to claims, and select a type when creating/editing a claim.

pedrobaeza commented on crm_claim_type/README.md:
```diff
@@ -0,0 +1,61 @@
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/crm/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/crm/issues/new?body=module:%20crm_claim_type%0Aversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
The link is not correctly formatted. Go to View mode of the file to see it.

osvalr replied:
It is not correctly formatted because it has **.md** as file extension as you commented [here](https://github.com/OCA/maintainer-tools/pull/94#issuecomment-127211142), but this is recognized when **.rst** is used instead. I renamed it to fix this

pedrobaeza commented on crm_claim_type/i18n/es.po:
```diff
@@ -0,0 +1,90 @@
+#. module: crm_claim_type
+#: model:ir.model,name:crm_claim_type.model_crm_claim
+msgid "Claim"
+msgstr "Reclamo"
```
Change "Reclamo" by "Reclamación" in all ocurrences, because this is how we called here in Spain.

nhomar replied:
@osvalr can you change Reclamación here, and put on es_XX Reclamo here in latinamérica Reclamación is not corret but it looks like in Spain it is.

pedrobaeza commented on crm_claim_type/models/crm_claim.py:
```diff
@@ -0,0 +1,37 @@
+
+    claim_type = \
+        fields.Many2one('crm.claim.type',
+                        help="Customer: When customer makes a claim to the company.\n"
```
Make it a more generic help, because you can create any typology.

nhomar replied:
Ok, This help is wired with the types you setted on data, can you simply explain here what type does. @osvalr

pedrobaeza commented on crm_claim_type/models/crm_claim_type.py:
```diff
@@ -0,0 +1,36 @@
+
+    name = fields.Char(required=True)
+    active = fields.Boolean(default=True)
+    description = fields.Text()
```
Make it translatable.

nhomar replied:
@osvalr translate=True in the field attribute in description and name

pedrobaeza commented on crm_claim_type/models/crm_claim_type.py:
```diff
@@ -0,0 +1,36 @@
+    """
+    _name = 'crm.claim.type'
+
+    name = fields.Char(required=True)
```
Make it translatable for companies with multi-language.

nhomar replied:
same than above.

pedrobaeza commented on crm_claim_type/security/ir.model.access.csv:
```diff
@@ -0,0 +1,3 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_crm_claim_type_user,crm.claim.type.user,model_crm_claim_type,base.group_sale_salesman,1,1,1,1
```
Salesman shouldn't change this. Only sales managers should.

nhomar replied:
+1 @osvalr

Commit:
[REF] crm_claim_type: correct installation, usage, bug tracker link and rst format

Commit:
[REF] crm_claim_type: replace help for claim_type field

Commit:
[REF] crm_claim_type: set translatable name and description

Commit:
[REF] crm_claim_type: change spanish term for claim

Commit:
[REF] crm_claim_type: change group id for security access rule

Commit:
[REF] crm_claim_type: fix readme bugtracker link

Commit:
[REF] crm_claim_type: fix readme bugtracker link

Commit:
[REF] crm_claim_type: fix readme bugtracker link

Commit:
Merge branch '8.0-oca-corrections-for-crm_claim_type-dev-osval' of github.com:vauxoo-dev/crm into 8.0-oca-corrections-for-crm_claim_type-dev-osval

Commit:
[REF] crm_claim_type: correct spanish terms and add new ones for es_MX and es_VE

nhomar committed:
Merge pull request #2 from vauxoo-dev/8.0-oca-corrections-for-crm_claim_type-dev-osval

8.0 oca corrections for crm claim type dev osval

nhomar commented:
@pedrobaeza all change applied.

pedrobaeza commented on crm_claim_type/README.rst:
```diff
@@ -0,0 +1,61 @@
+
+To use this module, you need to:
+
+* Go to Sales > Configurations > Claims > Claim Types
```
This line is already on Configuration. It's not needed here.

pedrobaeza commented on crm_claim_type/README.rst:
```diff
@@ -0,0 +1,61 @@
+* Go to Sales > Configurations > Claims > Claim Types
+* Then, go to Claims, and select a claim type when creating/editing a claim
+
+For further information, please visit:
```
You can put here the new cool "try me on runbot" button

```
.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
   :alt: Try me on Runbot
   :target: https://runbot.odoo-community.org/runbot/111/8.0
```

pedrobaeza replied:
Amended the repo id (i got confused with rma)

pedrobaeza commented on crm_claim_type/__init__.py:
```diff
@@ -0,0 +1,26 @@
+###############################################################################
+#    Module Writen to OpenERP, Open Source Management Solution
+#    Copyright (C) OpenERP Venezuela (<http://www.vauxoo.com>).
+#    All Rights Reserved
```
This sentence doesn't apply since long time ago

pedrobaeza commented on crm_claim_type/__openerp__.py:
```diff
@@ -0,0 +1,43 @@
+###############################################################################
+#    Module Writen to OpenERP, Open Source Management Solution
+#    Copyright (C) OpenERP Venezuela (<http://www.vauxoo.com>).
+#    All Rights Reserved
```
This sentence doesn't apply since long time ago

pedrobaeza commented on crm_claim_type/i18n/es.po:
```diff
@@ -0,0 +1,88 @@
+
+#. module: crm_claim_type
+#: field:crm.claim.type,description:0
+msgid "Decription"
```
s/Decription/Description

pedrobaeza commented on crm_claim_type/i18n/crm_claim_type.pot:
```diff
@@ -0,0 +1,89 @@
+
+#. module: crm_claim_type
+#: field:crm.claim.type,description:0
+msgid "Decription"
```
s/Decription/Description

pedrobaeza commented on crm_claim_type/security/ir.model.access.csv:
```diff
@@ -0,0 +1,3 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_crm_claim_type_user,crm.claim.type.user,model_crm_claim_type,base.group_sale_manager,1,1,1,1
```
salesman should have read access to this, or you will find a lot of errors.

pedrobaeza commented on crm_claim_type/README.rst:
```diff
@@ -0,0 +1,61 @@
+===============
+
+Includes a way to classify claims adding the concept of type.
+It gives a section to define a set of claim types.
```
I can't find anything about sections in the code. What do you mean with this?

yaniaular replied:
@pedrobaeza What word in the paragraph you mean exactly?

pedrobaeza replied:
The sections. You don't have any "section" in the code.

Commit:
[ADD] crm_claim_type: inherit crm claim form view

Commit:
[FIX] crm_claim_type: add permissions both groups technical features and salesman

Commit:
[REF] crm_claim_type: add license badget and new button, and remove useless line from Usage section

Commit:
[REF] crm_claim_type: correct typos in terms

rafaelbn commented:
@nhomar could you please show my difference between this module and Odoo's standard module in CRM? Odoo has already Type for claims (categ_id in crm.claim).

Commit:
[REF] crm_claim_type: update license header agreements

Commit:
[REF] crm_claim_type: update readme file

Commit:
[REF] The demo of crm_claim has claim_type like selection field, and in this module is changed to many2one field, then the value of claim_type in the demo data changes

nhomar committed:
Merge pull request #3 from vauxoo-dev/8.0-oca-corrections-crm_claim_type-dev-osval

8.0 oca corrections crm claim type dev osval

Commit:
[REF] guidelines applied in view

Commit:
[FIX] bad name in __OpenERP__.py about xml file

nhomar committed:
Merge pull request #4 from vauxoo-dev/8.0-oca-corrections-crm_claim_type-dev-yani

[REF] guidelines applied in view

nhomar commented:
Hello @rafaelbn 

crm.case.categ has a field called SECTION_ID, there is some reasons because we didn't use it:
1. There is a domain Odoo for categ_id crm.claim field where SECTION_ID comparing the claim with the category, forcing Display only one group of categories in the claim. And SECTION_ID crm.claim field is filled with the called user default_section_id field.
2. And the types of crm claims for customers, supplier or other, should be viewed by the customer user belongs regardless of whether or not a SECTION_ID. So I think they are two different things.
3. The default categories, which are already loaded Odoo `Factual Claims`, `Policy Claims`, `Value Claims`. Then, there may be a Customer complaint, but that you can have a Cutomer Claim which is `Factural Claims` and a Supplier Claim type that belongs to `Factural Claims` category also.
4. Then there would add value to say, the people of the Marketing team, or Direct Sales can handle certain categories of claims, but can be of different types (because any of them can have diferent workflows even).
5. Another reason Besides in the OCA field claim_type already it existed, only that it was a simple selection and we convert it many2one, if more types of claim, regardless of the category, which I think is something else was needed.
6. Another reason, is introduced with the aim to create different workflows for the company.
7. If there are three types claim, supplier, customer and other, each can have its workflow with different stages. 
8. In this case the field claim_Type crm.claim.stage is introduced into the model so that when creating a claim type of customer, it is only can display for the customer stages claim_type,
   so, the company, depending on your needs, you can create many types of claims as
   He wants to different stages, if you have various types of transitions.

I hope it help you to clarify.

Thanks.

rafaelbn commented:
Hi @nhomar ,

Thanks for you answer. To resume for other users, you add type to gain a level while classifying claims. This could give you more possibilities in the future (like point 6/7). Let's say I **could** be necessary for ISO/IEC 20000.
- In the claims you have the partner, so you know if it is from customer or supplier or neither.
- In the clams you have the reference so you know if is related with:
  - Partner
  - Product
  - Project
  - Project task
  - Invoice
  - Serial Number
  - Voucher
  - Delivery Order
  - Purchase Order
  - Sales Order
- So you can know if the claim is from an invoices from customer or supplier.

Nonetheless, if you introduce such a field don't you think you must create:

(1) add column in the list view,
(2) add a "group by" and
(3) add this also in Reporting -> Sales -> "Claim Analysis" (crm.claim.report)

of this field?

Here come a problem we must address (in my opinion an Odoo error - It looks like they set labels randomly...): Odoo names:
- In Sales -> After-Sales Service -> Claims: **Type** (in the view and in the groups by) / **Category** in the form view...
- In Reporting -> Sales -> Claims Analysis: **Category**

Conclusion: 1 field - 2 labels

When installing this module will be even more confused... I could propose to name this fields "Claim sort" or something similar? Do you think we can get something different to clarify this? (just labels could be enough)

Let me know you opinion

Thanks

yaniaular commented:
@rafaelbn 
1. With respect to the first points, a partner can be both: a customer and a supplier.
2. I think that to calculate if the claim is supplier type or customer type based in all information that you mentioned is more complicated. Let explain it field by field:
3. Partner: A partner can be both: a customer and a supplier.
4. Product: A product can be sold and can be purchased
5. Project and Project task: The claim does not relate to a project or project task
6. Invoice:  Ok, an invoice can be for a customer or from a supplier, nonetheless, the claim type is not a simple description, also affects the workflow of the claim. 

Like said @nhomar:

Another reason, is introduced with the aim to create different workflows for the company.
If there are three types claim, supplier, customer and other, each can have its workflow with different stages.

The crm.claim.stage model into crm_claim_rma module also will have the claim_type field. This means that a the claims can have a variety of workflows, as many as the company wants. 
[![See Video](http://img.youtube.com/vi/XlL7uVBlrtM/0.jpg)](https://youtu.be/XlL7uVBlrtM).

Depending of claim_type of the claim, it will display a certain workflow.

I think it is not necessary to continue explaining field by field, after watching the video.

(1) add column in the list view:     I agree with this
(2) add a "group by" and:  I agree with this

rafaelbn commented:
Hi @yaniaular , thanks for video!

Where do you configure the diferent stages for each Claim Type?

I have checked Sales -> Configuration -> Claim -> Claim types / Stages.

Thanks
Regards

osvalr commented:
Hi @rafaelbn , the crm_claim_rma module enables this particular configuration section that you're asking about, so in order to have it working and configure it as well your have to install crm_claim_rma module.

By the way, we're thinking to move crm.claim.stage inherited model from crm_claim_rma module to crm_claim_type, what are your thoughts about it?

Commit:
[ADD] crm_claim_type: add crm_claim_stage based on crm_claim_type

crm_claim_stage functionality was moved from crm_claim_rma to crm_claim_type to
fulfill crm_claim_type module function, also aditional data and demo was added

Commit:
[FIX] crm_claim_type: remove already deleted file entry from __openerp__

Commit:
[REF] cmr_claim_type: add missing terms to i18n

Commit:
[REF] crm_claim_type: add visual logic and switch field positions

Commit:
[REF] crm_claim_type: remove redundant field stage_name_uniq from module

Commit:
[REF] crm_claim_type: add group to claim_type

Commit:
[REF] crm_claim_type: extend module description in README file

Commit:
[REF] crm_claim_type: move claim_type field

Commit:
[REF] crm_claim_type: unset translatable to claim_type

Commit:
[REF] crm_claim_type: add demos, override extending data for crm_claim

Commit:
[REF] crm_claim_type: add EOF

Commit:
[REF] crm_claim_type: update i18n terms

nhomar committed:
Merge pull request #5 from vauxoo-dev/8.0-crm_claim_stage-to-crm_claim_type-dev-osval

8.0 crm claim stage to crm claim type dev osval

yaniaular commented:
@pedrobaeza, @rafaelbn and @nhomar, do you agree with the changes applied?

pedrobaeza commented:
You have 46 commits for a newcomer module. You should squash them all together.

yaniaular commented:
This PR (https://github.com/OCA/crm/pull/39) is replaced by https://github.com/OCA/crm/pull/43. This PR must be closed.  @pedrobaeza
